### PR TITLE
Fix SplitException

### PR DESCRIPTION
**Pull request by bchong95**
# Fix SplitException

## Description

CosmosQueryClientCore did not implement TryGetOverlappingRangesAsync, so when a query tries split proofing it runs into a NotImplementedException. We didn't have any integration tests (nor do we have any now), since the public emulator can't simulate a split.

## Type of change

Please delete options that are not relevant.

- [x ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by kirankumarkolli** [comment]
@bchong95 thoughts on how to cover split tests?

File: changelog.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 - [#1070](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1070) CreateItem will only retry for auto-extracted partition key in-case of collection re-creation
 - [#1060](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1060) Fixed unicode encoding bug in DISTINCT queries.
+- [#1091](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1091) Fixed NotImplementedException in for splits.
```
Comment (kirankumarkolli) [inline changelog.md]: Better wording about the bug and its impact will help CX. Current description seems doesn't convey that even working queries might run into this issue.

Comment (j82w) [reply changelog.md]: How about:
"Fixed a bug in query when a partition split occurs that causes a NotImplementedException to be thrown."

Comment (bchong95) [reply changelog.md]: done.

**Pull request merged**